### PR TITLE
feat: allow to serve hidden files

### PR DESCRIPTION
**Pull request by leandroandrade**
The purpose of this PR is to help with issue #303 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -336,7 +336,7 @@ async function fastifyStatic (fastify, opts) {
       const winSeparatorRegex = new RegExp(`\\${path.win32.sep}`, 'g')
 
       for (const rootPath of Array.isArray(sendOptions.root) ? sendOptions.root : [sendOptions.root]) {
-        const files = await globPromise(path.join(rootPath, globPattern).replace(winSeparatorRegex, path.posix.sep), { nodir: true })
+        const files = await globPromise(path.join(rootPath, globPattern).replace(winSeparatorRegex, path.posix.sep), { nodir: true, dot: true })
```
Comment (jsumners) [inline index.js:339]: The should be configurable, and it _definitely_ should not default to `true`.

**Review by jsumners — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (leandroandrade) [reply index.js:339]: Thanks. I really appreciate your feedback.

File: index.js [hunk 2]
```diff
@@ -37,7 +37,8 @@ async function fastifyStatic (fastify, opts) {
     immutable: opts.immutable,
     index: opts.index,
     lastModified: opts.lastModified,
-    maxAge: opts.maxAge
+    maxAge: opts.maxAge,
+    serveHiddenFiles: opts.serveHiddenFiles
```
Comment (jsumners) [inline index.js:41]:
```suggestion
    serveHiddenFiles: opts.serveHiddenFiles ?? false
```

File: README.md [hunk 3]
```diff
@@ -201,6 +201,12 @@ Default: `undefined`
 Under the hood we use [send](https://github.com/pillarjs/send#index) lib that by default supports "index.html" files. 
 To disable this set false or to supply a new index pass a string or an array in preferred order.
 
+### `serveHiddenFiles`
+
+Default: `undefined`
```
Comment (jsumners) [inline README.md:206]:
```suggestion
Default: `false`
```

File: README.md [hunk 4]
```diff
@@ -201,6 +201,12 @@ Default: `undefined`
...
+
+This option allow to serve files into hidden directory using `wildcard` as `false`.
```
Comment (jsumners) [inline README.md:208]:
```suggestion
When `true`, files in hidden directories (e.g. `.foo`) will be served.
```

File: index.js (hunk 2 above)
Comment (leandroandrade) [reply index.js:41]: Added suggestions. Thanks.

File: README.md (hunk 3 above)
Comment (leandroandrade) [reply README.md:206]: Added suggestions. Thanks.

File: README.md (hunk 4 above)
Comment (leandroandrade) [reply README.md:208]: Added suggestions. Thanks.

**Comment by Uzlopak** [comment]
We should imho name the option serverDotFiles . Hidden files in windows are files which have the hidden flag set. Probably in Linux the same. 

So in strongly suggest to rename the option.

**Comment by leandroandrade** [comment]
Hi @Uzlopak , you are right.

**Comment by Uzlopak** [comment]
I apologize, i made a typo. Should be serveDotFiles and not serverDotFiles. Can you please fix that?

File: README.md [hunk 5]
```diff
@@ -201,7 +201,7 @@ Default: `undefined`
 Under the hood we use [send](https://github.com/pillarjs/send#index) lib that by default supports "index.html" files. 
 To disable this set false or to supply a new index pass a string or an array in preferred order.
 
-### `serveHiddenFiles`
+### `serverDotFiles`
```
Comment (mcollina) [inline README.md:204]:
```suggestion
### `serveDotFiles`
```

**Review by mcollina — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (leandroandrade) [reply README.md]: Changes made.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by Uzlopak — APPROVED** [review]

**Comment by Uzlopak** [comment]
@jsumners PTAL

**Review by jsumners — APPROVED** [review]

**Comment by jsumners** [comment]
> @jsumners PTAL

Please have patience. We live several time zones apart.

**Pull request merged**